File: OCH_Tao/ABC/B/027.py
N = int(input())
A = list(map(int,input().split()))
cnt = 0
if sum(A)%N != 0:
  print(-1)
else:
  x = sum(A)//N
  l = []
  for i in range(N):
    l.append(A[i]-x)
  for i in range(N):
    if l[i]==0:
      cnt += 1
      continue
    else:
      while l[i] < 0:
        l[i] += 1
        l[i+1] -= 1
      while l[i] > 0:
        l[i] -= 1
        l[i+1] += 1
  else:
    print(N-cnt)

File: OCH_Tao/ABC/B/027.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const N = parseInt(input[0], 10);
const A = (input[1].split(" ")).map(x => parseInt(x, 10));
let cnt = 0;
if (A.reduce((x, y) => x + y) % N === 0) {
  const one = A.reduce((x, y) => x + y) / N;
  for (let i = 1; i < N; i++) {
    let l = A.slice(0, i);
    if (l.reduce((x, y) => x + y) !== one * i) {
      cnt++;
    }
  }
} else {
  cnt = -1;
}
console.log(cnt);